### DISCUSSION TITLE
Providing updated translations continously

### DISCUSSION THREAD

**Discussion by stefonarch** [lxqt/lxqt#2292] [Ideas]
I noticed that always immediately _after_ the new release there is a peak in new translations coming in.

As it's possible to add translated `*.qm` and `*.desktop` files  in `~/.local/share/lxqt/translations` ,  `~/.local/share/applications/` ,  `~/.local/share/pcmanfm-qt/translations/`  and so on we could provide them on a weekly or daily base in GH or on the website so translators and other interested people could benefit from. 

For some reason `~/.local/share/desktop-directories/` is not considered by the system, I  noticed. Except for those this system doesn't interfere with packages from distros.

**Comment by tsujan** [comment]
Users could have updated translations only when they update the apps. A long time ago, translations might have had a separate section (did they?). Having translations of each component/app in its own source seems the cleanest and best approach to me, especially because LXQt is modular.

**Comment by tsujan** [comment]
> For some reason ~/.local/share/desktop-directories/ is not considered by the system

It is here. A long time ago, I created desktop entries for Wine directories there, and they show up under Main Menu → Wine.

**Comment by tsujan** [comment]
> ~/.local/share/pcmanfm-qt/translations/

I don't think it'll work. pcmanfm-qt loads its translations from `${CMAKE_INSTALL_PREFIX}/share/pcmanfm-qt/translations` — other apps do similarly.

**Comment by stefonarch** [comment]
Tested, I remebered wrongly, it doesn't work. So root privileges would be needed to save those *.qm files for lximage-qt, pcmanfm-qt and lxqt-archiver.

With `~/.local/share/desktop-directories/` I tested this morning with `lxqt-leave.directory` and it didn't work, but will recheck.

**Reply by tsujan** [reply]
> With ~/.local/share/desktop-directories/ I tested this morning

I have `~/.local/share/desktop-directories/wine-wine.directory` as:
```
[Desktop Entry]
Type=Directory
Name=Wine
Icon=wine
```
And `~/.local/share/desktop-directories/wine-Programs.directory` as:
```
[Desktop Entry]
Type=Directory
Name=Programs
Icon=folder-software
```
I also have the directories `~/.local/share/applications/wine/` and `~/.local/share/applications/wine/Programs/`.

But, most importantly, I have `~/.config/menus/applications-merged/wine-Programs-all.menu` as:
```
<!DOCTYPE Menu PUBLIC "-//freedesktop//DTD Menu 1.0//EN"
"http://www.freedesktop.org/standards/menu-spec/menu-1.0.dtd">
<Menu>
  <Name>Applications</Name>
  <Menu>
    <Name>wine-wine</Name>
    <Directory>wine-wine.directory</Directory>
    <Include>
      <Category>Wine</Category>
    </Include>
    <Menu>
      <Name>wine-Programs</Name>
      <Directory>wine-Programs.directory</Directory>
      <Include>
        <Category>WineProgram</Category>
      </Include>
    </Menu>
  </Menu>
</Menu>
```

EDIT: Restarting of lxqt-panel may be needed.

**Comment by stefonarch** [comment]
I tested again: 
1. copied `/usr/share/desktop-directories/lxqt-leave.directory`  to `~/.local/share/desktop-directories/`
2. Edited the italian entry
3. restarted panel, restarted session: no changed text is seen.

Found the reason now: removing the original file it works, but files  present in `/usr/share/desktop-directories/`  override   local ones here.

**Reply by tsujan** [reply]
> files present in /usr/share/desktop-directories/ override local ones here

I'm not sure if the following is related, but quoting from https://specifications.freedesktop.org/menu-spec/menu-spec-1.0.html, under `<DefaultDirectoryDirs>`:

"...The default locations that are earlier in the search path go later in the `<Menu>` so that they have priority."

~`/usr/share/desktop-directories/` may be searched first when the menu file is in `/etc/xdg/menus/`; just a guess, nothing more.~

**Reply by tsujan** [reply]
No, my guess was wrong. `~/.local/share/desktop-directories/` has priority, as it does with pcmanfm-qt → Applications (you may need to wait a few seconds to see it). Apparently, LXQt has a small problem in this regard. Probably, you could open an issue at `libqtxdg` (not completely sure about the place but `libqtxdg` seems right).

**Reply by tsujan** [reply]
No need to open an issue: https://github.com/lxqt/libqtxdg/pull/285

**Comment by stefonarch** [comment]
A draft is here: https://github.com/stefonarch/LXQt_translations_update/ 
I tested it in VBox with Lubuntu 22.04 and it seems working fine. 
Credits to @ali-alhaidary for the idea.

On a side note: all lxqt-components can have their local translation except lxqt-panel (if I tested correctly).
